tests/foreman/ui/test_location.py

```diff
@@ -524,10 +528,18 @@ def test_positive_remove_environment(self):
             for env_name in valid_env_names():
                 with self.subTest(env_name):
                     loc_name = gen_string('alpha')
-                    env = entities.Environment(name=env_name).create()
+                    env = entities.Environment(
+                        name=env_name,
+                        organization=[self.org_],
+                    ).create()
                     self.assertEqual(env.name, env_name)
-                    set_context(session, org=ANY_CONTEXT['org'])
-                    make_loc(session, name=loc_name, envs=[env_name])
+                    set_context(session, org=self.org_.name)
```

Comment:
Just for curiosity:

**set_context** call takes 7 seconds

```python
time.ctime(); set_context(session, org='fyxZQlB'); time.ctime()
'Thu Jun  2 13:04:01 2016'
2016-06-02 13:04:01 - robottelo.ui.locators - DEBUG - Accessing locator 
.... 10 hidden operations...... 
'Thu Jun  2 13:04:08 2016'
```

While crating a new user with a default_organization takes only 1 second

```python
time.ctime(); entities.User(default_organization=self.org_.id).create(); time.ctime()
'Thu Jun  2 13:06:09 2016'
2016-06-02 13:06:09 - nailgun.client - DEBUG - Making HTTP POST request to ...
'Thu Jun  2 13:06:10 2016'
```

So maybe we can create a new user in setup using that organization as default, pass the user to **with Session** and avoid calls to set_context qhen it is only for organization?